FirebaseStorage/Sources/Internal/StorageDeleteTask.swift

```diff
@@ -44,34 +44,34 @@ internal class StorageDeleteTask: StorageTask, StorageTaskManagement {
    * Prepares a task and begins execution.
    */
   internal func enqueue() {
-    weak var weakSelf = self
-    dispatchQueue.async {
-      guard let strongSelf = weakSelf else { return }
-      strongSelf.state = .queueing
-      var request = strongSelf.baseRequest
+    let completion = taskCompletion
+    taskCompletion = { (error: Error?) in
+      completion?(error)
+      // Reference self in completion handler in order to retain self until completion is called.
+      self.taskCompletion = nil
+    }
```

Comment:
Instead of doing this, why not just let the `dispatchQueue.async` closure capture `self` strongly? (Here and throughout)

Reply:
I tried that, but it's not sufficient for self to get deallocated before entering outer `enqueue` `dispatchAsync`

Reply:
I _think_ that would create a retain-cycle because `self` has a strong reference to `dispatchQueue` in [`StorageTask`](https://github.com/firebase/firebase-ios-sdk/blob/3d51280713332af3f023abc7e305541871c1e005/FirebaseStorage/Sources/StorageTask.swift#L80) and `dispatchQueue` would then hold a strong reference back to `self` as well.

Reply:
Here's the backtrace from the `deinit`: 

![Screen Shot 2022-11-23 at 3 56 38 PM](https://user-images.githubusercontent.com/73870/203665571-b32b4763-2b0f-4210-a6ab-502b16e4ebe8.png)